### PR TITLE
New damage plugin

### PR DESCRIPTION
**Pull request by BhaalM**
Ported from Baaleos nwnx2 damage plugin.

### CODE REVIEW

File: Plugins/Damage/Damage.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
+
+// Log currently generates warnings when no arguments are given to format string
+// TODO: Should really clean up the log so it doesn't warn in these cases
+#pragma GCC diagnostic ignored "-Wformat-security"
```
Comment (mtijanic) [inline Plugins/Damage/Damage.cpp]: This can be removed now :)

**Comment by BhaalM** [comment]
I've seen that the event plugin is usually for shared hooks this one require exclusive one to change the damage values... I thought this was not the idea for the Event plugin. But, as you wish.

The problem with the damage event is that you cannot change the damage inflicted to a creature. You can heal a creature after been damaged, or apply a new bonus damage... but you cannnot directly set the amount of damage or what is displayed on the combat log.

**Comment by mtijanic** [comment]
Ack on both accounts. I still think it belongs in events, but it would actually require extending that plugin to support passing the data back (I'm not sure if NWNX_Events_PushEventData() would cover it).

**Comment by Christian-Vassallo** [comment]
It belong to nwnx_damages.

It is different because it allows to modify the damages and we can think it as a layer between the system and the creature (when the creature get damaged).

Nwnx events instead is a lauer between the creature and the system (when you fire an action)

Moreover, in my opinion, for a back-compatibility of concepts, nwnx_damages should stay as a plugin separated :)

File: Plugins/Damage/Damage.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+namespace Damage {
+
+class Damage : public NWNXLib::Plugin
+{
+   	public:
+    	Damage(const Plugin::CreateParams& params);
+      	virtual ~Damage();
+
+   	private:      
+   		ArgumentStack SetDamageEventScript(ArgumentStack&& args);
```
Comment (Liareth) [inline Plugins/Damage/Damage.hpp]: These should be spaces rather than tabs.

**Comment by Liareth** [comment]
I think this would be better as an event with the ability to set the damage value that goes through, but that would require changing the events plugin, and to be honest ... where it is is much less important than if it is. If we ever decide to make it an even we can just deprecate this.

Awesome work @BhaalM!

**Pull request merged**